File: engine/contracts/operations.py
"""此文件由 scripts/generate-contracts.py 生成，请勿手改。"""
from __future__ import annotations

OPERATION_PLAN_ID_PREFIX = 'op_'
OPERATION_KINDS = frozenset(('edit', 'migration', 'metadata', 'delete', 'restore-delete'))
EDIT_OPERATION_KINDS = frozenset(('delete-turn', 'rewrite', 'replace-assistant-reply'))
OPERATION_STATUSES = frozenset(('planned', 'queued', 'applying', 'applied', 'failed', 'cancelled', 'expired'))
OPERATION_TERMINAL_STATUSES = frozenset(('applied', 'failed', 'cancelled', 'expired'))
OPERATION_SUCCESS_STATUS = 'applied'

File: app/src-tauri/src/contracts/operations.rs
// 此文件由 scripts/generate-contracts.py 生成，请勿手改。
pub(crate) const OPERATION_PLAN_ID_PREFIX: &str = "op_";
pub(crate) const OPERATION_KINDS: &[&str] =
    &["edit", "migration", "metadata", "delete", "restore-delete"];
pub(crate) const EDIT_OPERATION_KINDS: &[&str] =
    &["delete-turn", "rewrite", "replace-assistant-reply"];
pub(crate) const OPERATION_STATUSES: &[&str] = &[
    "planned",
    "queued",
    "applying",
    "applied",
    "failed",
    "cancelled",
    "expired",
];
pub(crate) const OPERATION_TERMINAL_STATUSES: &[&str] =
    &["applied", "failed", "cancelled", "expired"];
pub(crate) const OPERATION_SUCCESS_STATUS: &str = "applied";
